nominatim/tokenizer/sanitizers/config.py

```diff
@@ -102,30 +106,41 @@ def get_delimiter(self, default: str = ',;') -> Pattern[str]:
         return re.compile('\\s*[{}]+\\s*'.format(''.join('\\' + d for d in delimiter_set)))
 
 
-    def get_filter_kind(self, *default: str) -> Callable[[str], bool]:
-        """ Return a filter function for the name kind from the 'filter-kind'
-            config parameter.
+    def get_filter(self, param: str, default: Optional[Sequence[str]] = None
+            ) -> Callable[[Optional[str]], bool]:
+        """ Returns a filter function for the given parameter of the sanitizer
+            configuration.
 
-            If the 'filter-kind' parameter is empty, the filter lets all items
-            pass. If the parameter is a string, it is interpreted as a single
-            regular expression that must match the full kind string.
-            If the parameter is a list then
-            any of the regular expressions in the list must match to pass.
+            The value provided for the parameter in sanitizer configuration
+            should be a string or list of strings, where each string is a regular
+            expression. These regular expressions will later be used by the
+            filter function to filter strings.
 
             Arguments:
-                default: Filters to be used, when the 'filter-kind' parameter
-                         is not specified. If omitted then the default is to
-                         let all names pass.
+                param: The parameter for which the filter function
+                       will be created.
+                default: Takes a list of strings where each string
+                         is a regular expression.These regular expressions will
+                         be used by the filter function in case the given parameter
+                         is missing in the sanitizer configuration.
+                         Note that if this default is not provided, the filter function
+                         lets all target strings pass, including None type.
 
             Returns:
-                A filter function which takes a name string and returns
-                True when the item passes the filter.
+                A filter function that takes a target string as the argument and
+                returns True if the it fully matches any of the regular expressions
+                otherwise returns False.
+                Note that if instead of target string, None type is provided then
+                it simply returns False.
         """
-        filters = self.get_string_list('filter-kind', default)
+        values = self.data.get(param, None)
+        filters = self.get_string_list(param) if values is not None else default
```

Comment:
What is the motivation to change the semantics of `get_string_list()`? It just makes the code here a lot more complicated. Without the change, you can simply leave the code here untouched and it works.

Reply:
Actually, here we want to check if both `values` and `default` are `none` so I am explicitly checking that. With the current definition of `get_string_list`, it should not return `none`, because type checking would complain.

 And in case we allow it to also return `none`, we would explicitly have to check for `none` values wherever we use it. Also, I think it would be good idea to let the function only return a list as it would make the function return value be more predictable.

Reply:
That could be done, but I would prefer to be it like this.